helix-core/src/main/java/org/apache/helix/util/HelixUtil.java

```diff
@@ -295,7 +352,6 @@ public static String serializeByComma(List<String> objects) {
       LOG.error("getIdealAssignmentForWagedFullAuto(): Failed to compute ResourceAssignments!", e);
     } finally {
       // Close all ZK connections
-      baseDataAccessor.close();
       readOnlyWagedRebalancer.close();
```

Comment:
readOnlyWagedRebalancer is also passed in object, why closing it but removing baseDataAccessor close?

Reply:
If we pass a baseDataAccessor created by caller, then that accessor should not be closed here. readOnlyWagedRebalancer is created in this function so we close it in final block.

Reply:
For helix-reset, the caller create the accessor and passed in, right? So it is not created in this function.

Reply:
Yes this is correct. So we should not close accessor here.

Reply:
I see, I was thinking about the zkBucketDataAccessor. It is not closed when the rebalancer is closed.
You are right.